WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+SiteIcon.swift

```diff
@@ -3,73 +3,79 @@ import WordPressFlux
 import WordPressShared
 import SwiftUI
 import SVProgressHUD
+import Gridicons
+import PhotosUI
 
 extension SitePickerViewController {
 
-    func showSiteIconSelectionAlert() {
-        let alert = UIAlertController(title: SiteIconAlertStrings.title,
-                                      message: nil,
-                                      preferredStyle: .actionSheet)
-
-        alert.popoverPresentationController?.sourceView = blogDetailHeaderView.blavatarImageView.superview
-        alert.popoverPresentationController?.sourceRect = blogDetailHeaderView.blavatarImageView.frame
-        alert.popoverPresentationController?.permittedArrowDirections = .any
-
-        alert.addDefaultActionWithTitle(SiteIconAlertStrings.Actions.chooseImage) { [weak self] _ in
-            NoticesDispatch.unlock()
-            self?.updateSiteIcon()
-        }
-
-        alert.addDefaultActionWithTitle(SiteIconAlertStrings.Actions.createWithEmoji) { [weak self] _ in
-            NoticesDispatch.unlock()
-            self?.showEmojiPicker()
+    func makeSiteIconMenu() -> UIMenu? {
+        guard siteIconShouldAllowDroppedImages() else {
+            return nil
         }
-
-        alert.addDestructiveActionWithTitle(SiteIconAlertStrings.Actions.removeSiteIcon) { [weak self] _ in
-            NoticesDispatch.unlock()
-            self?.removeSiteIcon()
-        }
-
-        alert.addCancelActionWithTitle(SiteIconAlertStrings.Actions.cancel) { [weak self] _ in
-            NoticesDispatch.unlock()
-            self?.startAlertTimer()
-        }
-
-        present(alert, animated: true)
+        return UIMenu(children: [
+            UIDeferredMenuElement.uncached { [weak self] in
+                $0(self?.makeUpdateSiteIconActions() ?? [])
+            }
+        ])
     }
 
-    func showUpdateSiteIconAlert() {
-        let alert = UIAlertController(title: nil,
-                                      message: nil,
-                                      preferredStyle: .actionSheet)
-
-        alert.popoverPresentationController?.sourceView = blogDetailHeaderView.blavatarImageView.superview
-        alert.popoverPresentationController?.sourceRect = blogDetailHeaderView.blavatarImageView.frame
-        alert.popoverPresentationController?.permittedArrowDirections = .any
-
-        alert.addDefaultActionWithTitle(SiteIconAlertStrings.Actions.changeSiteIcon) { [weak self] _ in
-            NoticesDispatch.unlock()
-            self?.updateSiteIcon()
-        }
-
-        if blog.hasIcon {
-            alert.addDestructiveActionWithTitle(SiteIconAlertStrings.Actions.removeSiteIcon) { [weak self] _ in
-                NoticesDispatch.unlock()
-                self?.removeSiteIcon()
+    func didShowSiteIconMenu() {
```

Comment:
There doesn't seem to be any good way to track `UIMenu` dismiss event, so I implemented this "pull-based" check for the site icon flow termination. Not ideal, but works.